File: server/src/models/Recipe.ts
import mongoose from "mongoose";

const { Schema } = mongoose;

export interface RecipeDocument extends mongoose.Document {
  title: string;
  description: string;
  time: string;
  ownerId?: string;
  user?: string;
  cusine: string;
  utensils: string;
  category: string;
  createdAt?: string;
  image?: string;
  instructions: string;
  ingredients?: [
    {
      item: string;
      quantity: string;
    }
  ];
  review?: [{}];
}

const RecipeSchema: mongoose.Schema = new Schema({
  time: String,
  ownerId: Schema.Types.ObjectId,
  user: String,
  cusine: String,
  utensils: String,
  title: String,
  image: String,
  createdAt: {
    type: Date,
    default: Date.now,
  },
  description: String,
  instructions: String,
  ingredients: [
    {
      item: String,
      quantity: String,
    },
  ],
  reviews: [
    {
      type: Schema.Types.ObjectId,
      ref: "Review",
    },
  ],
  category: String,
});

const Recipe: mongoose.Model<RecipeDocument> = mongoose.model(
  "Recipe",
  RecipeSchema
);

export default Recipe;

File: lib/model/recipe.dart
class Recipe{
  String url;
  String title;
  String description;
  String postid;
  String ownerid;
  String time;
  String ingredients;
  String user;
  String cusine;
  String utensils;

  Recipe({this.title, this.ownerid, this.user, this.postid, this.ingredients, this.url, this.description, this.time, this.cusine, this.utensils});

  Recipe.fromData(Map<String, dynamic> json)
  : url = json["url"],
  title = json["title"],
  postid = json["postid"],
  user = json["user"],
  description = json["description"],
  time = json["time"],
  ownerid = json["ownerid"],
  ingredients = json["ingredients"],
  cusine = json["cusine"],
  utensils = json["utensils"];

  Map<String, dynamic> toJson(){
    return{
      "url" : url,
      "title" : title,
      "user" : user,
      "description" : description,
      "time" : time,
      "postid" : postid,
      "ownerid" : ownerid,
      "ingredients" : ingredients,
      "cusine" : cusine,
      "utensils" : utensils
    };
  }
}